src/dfuhelper.c

```diff
@@ -61,7 +61,7 @@ int connected_normal_mode(const usbmuxd_device_info_t *usbmuxd_device) {
 		LOG(LOG_ERROR, "Unable to get device information");
 		return 0;
 	}
-	if (strncmp(dev.CPUArchitecture, "arm64", strlen("arm64"))) {
+	if (strncmp(dev.CPUArchitecture, "arm64", strlen(dev.CPUArchitecture))) {
```

Comment:
Since `dev.CPUArchitecture` will never be shorter than 5 characters, this call is equivalent to `strcmp(dev.CPUArchitecture, "arm64")`.

Reply:
what even is the point in this

Reply:
🗣🗣🗣🔥🔥🔥🗣🗣

Reply:
there is no point

Reply:
> what even is the point in this

The first one will evaluate to true for arm64, arm64e, and arm64_32 because it only checks the first 5 characters, and palera1n doesn't support arm64e (A12) and arm64_32 (Apple Watch). Either way, what even is the point of using strncmp when you can use strcmp